source/simulator/checkpoint_restart.cc

```diff
@@ -96,10 +111,127 @@ namespace aspect
+      AssertThrow (use_locally_conservative_discretization == parameters.use_locally_conservative_discretization,
+                   ExcMessage ("The value provided for `Use locally conservative discretization' that was stored "
+                               "in the checkpoint file is not the same as the one "
+                               "you currently set in your input file. "
+                               "These need to be the same during restarting "
+                               "from a checkpoint."));
+
+      bool use_discontinuous_temperature_discretization;
+      ia >> use_discontinuous_temperature_discretization;
+      AssertThrow (use_discontinuous_temperature_discretization == parameters.use_discontinuous_temperature_discretization,
+                   ExcMessage ("The value provided for `Use discontinuous temperature discretization' that was stored "
+                               "in the checkpoint file is not the same as the one "
+                               "you currently set in your input file. "
+                               "These need to be the same during restarting "
+                               "from a checkpoint."));
+
+      bool use_discontinuous_composition_discretization;
+      ia >> use_discontinuous_composition_discretization;
+      AssertThrow (use_discontinuous_composition_discretization == parameters.use_discontinuous_composition_discretization,
+                   ExcMessage ("The value provided for `Use discontinuous composition discretization' that was stored "
+                               "in the checkpoint file is not the same as the one "
+                               "you currently set in your input file. "
+                               "These need to be the same during restarting "
+                               "from a checkpoint."));
+
+      unsigned int temperature_degree;
+      ia >> temperature_degree;
+      AssertThrow (temperature_degree == parameters.temperature_degree,
+                   ExcMessage ("The temperature polynomial degree that was stored "
+                               "in the checkpoint file is not the same as the one "
+                               "you currently set in your input file. "
+                               "These need to be the same during restarting "
+                               "from a checkpoint."));
+
+      unsigned int composition_degree;
+      ia >> composition_degree;
+      AssertThrow (composition_degree == parameters.composition_degree,
+                   ExcMessage ("The composition polynomial degree that was stored "
+                               "in the checkpoint file is not the same as the one "
+                               "you currently set in your input file. "
+                               "These need to be the same during restarting "
+                               "from a checkpoint."));
+
+      std::string pressure_normalization;
+      ia >> pressure_normalization;
```

Comment:
I think this is okay to be changed, at least the computation should still work.

Reply:
@tjhei -- I thought about that, but it will result in a simulation in which the pressure will likely have a drastic jump from one time step to the next. I don't think that's a useful mode of usage.